File: mobile/src/main/javaBackup/com/planit/home/planit007/MainActivity.java
package planit007.planit.home.planit007;

import android.content.ContentValues;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.planit.home.planit007.Adapters.EventAdapter;
import com.planit.home.planit007.Adapters.GroupAdapter;
import com.planit.home.planit007.data.Contracts.EventContract;
import com.planit.home.planit007.DbHelpers.EventDbHelper;
import com.planit.home.planit007.Contracts.GroupContract;
import com.planit.home.planit007.DbHelpers.GroupDbHelper;
import com.planit.home.planit007.Useful;
import com.github.clans.fab.FloatingActionMenu;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.MobileAds;
import com.planit007.Adapters.EventAdapter;
import com.planit007.Adapters.GroupAdapter;

import com.planit007.data.Contracts.EventContract;
import com.planit007.data.Contracts.GroupContract;
import com.planit007.data.DbHelpers.EventDbHelper;
import com.planit007.data.DbHelpers.GroupDbHelper;
import com.planit007.data.Useful;

import planit007.planit.home.planit007.data.Contracts.EventContract;
import planit007.planit.home.planit007.data.Contracts.GroupContract;
import planit007.planit.home.planit007.data.Useful;


//7.04 ONWARDS NEEDS TO BE DONE FOR EVENTS

public class MainActivity extends AppCompatActivity implements GroupAdapter.GroupAdapterOnClickHandler, EventAdapter.EventAdapterOnClickHandler {
    Context context = this;

    FloatingActionMenu famMain;

    private SQLiteDatabase mGroupDb;
    private SQLiteDatabase mEventDb;

    private GroupAdapter mGroupAdapter;
    private EventAdapter mEventAdapter;

    private RecyclerView mGroupsList;
    private RecyclerView mEventsList;

    private Cursor groupCursor;
    private Cursor eventCursor;

    private TextView txtAddGroup;
    private TextView txtAddEvent;

    private TextView txtNullGroup;
    private TextView txtNullEvent;

    private boolean famIsOpen = false;

    String TAG = "MainActivity";

    private AdView mAdView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(planit.planit.home.planit003.R.layout.activity_main);

        mAdView = findViewById(planit.planit.home.planit003.R.id.m_ad);
        AdRequest adRequest = new AdRequest.Builder().build();
        mAdView.loadAd(adRequest);

        MobileAds.initialize(this, "ca-app-pub-1144529456090164~2412831884");

        mGroupsList = (RecyclerView) findViewById(planit.planit.home.planit003.R.id.m_rv_groups);
        mEventsList = (RecyclerView) findViewById(planit.planit.home.planit003.R.id.m_rv_events);

        LinearLayoutManager groupLayoutManager = new LinearLayoutManager(this);
        LinearLayoutManager eventLayoutManager = new LinearLayoutManager(this);

        mGroupsList.setLayoutManager(groupLayoutManager);
        mEventsList.setLayoutManager(eventLayoutManager);

        GroupDbHelper groupDbHelper = new GroupDbHelper(this);
        EventDbHelper eventDbHelper = new EventDbHelper(this);

        mGroupDb = groupDbHelper.getReadableDatabase();
        mEventDb = eventDbHelper.getReadableDatabase();

        groupCursor = getAllGroups();
        eventCursor = getAllEvents();

        txtNullGroup = (TextView) findViewById(planit.planit.home.planit003.R.id.m_tv_groups_null);
        txtNullEvent = (TextView) findViewById(planit.planit.home.planit003.R.id.m_tv_events_null);

        if(!eventCursor.moveToFirst()) {

            Log.d(TAG, "DEBUG MA 77 EVENT CURSOR NULL");
        }

        mGroupsList.setHasFixedSize(true);
        mEventsList.setHasFixedSize(true);

        mGroupAdapter = new GroupAdapter(this, groupCursor, this);
        mEventAdapter = new EventAdapter(this, eventCursor, this);

        mGroupsList.setAdapter(mGroupAdapter);
        mEventsList.setAdapter(mEventAdapter);

        updateNullView(mGroupAdapter, txtNullGroup);
        updateNullView(mEventAdapter, txtNullEvent);

        famMain = (FloatingActionMenu) findViewById(planit.planit.home.planit003.R.id.m_fam);

        findViewById(planit.planit.home.planit003.R.id.m_fab_add_group).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {

                final Useful.NameAlertDialog groupNAD = new Useful.NameAlertDialog("Group", "CREATE", "CANCEL", "", new EditText(context), context);
                groupNAD.create();
                groupNAD.ad.setOnShowListener(new DialogInterface.OnShowListener() {
                    @Override
                    public void onShow(DialogInterface dialogInterface) {

                        groupNAD.setBtnColor();
                        groupNAD.ad.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
                            @Override
                            public void onClick(View view) {

                                if(groupNAD.et.getText().toString().equals("") || groupNAD.et.getText().toString().equals(null)) {

                                    Toast t = Toast.makeText(getApplicationContext(), "Please enter a group name", Toast.LENGTH_LONG);

                                    t.show();
                                }

                                else {

                                    Intent i = new Intent(MainActivity.this, GroupDetailsActivity.class);

                                    famMain.close(true);

                                    groupNAD.ad.dismiss();

                                    i.putExtra("groupName", groupNAD.et.getText().toString());
                                    i.putExtra("groupID", (int) (addNewGroup(groupNAD.et.getText().toString())));

                                    startActivity(i);
                                }
                            }
                        });
                    }
                });

                groupNAD.ad.show();
            }
        });

        findViewById(planit.planit.home.planit003.R.id.m_fab_add_event).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {

                if(mGroupAdapter.getItemCount() == 0) {

                    Toast t = new Toast(context);

                    t.makeText(context, "Please create a group before creating an event", Toast.LENGTH_SHORT).show();
                }

                else {

                    final Useful.NameAlertDialog eventNAD = new Useful.NameAlertDialog("Event", "CREATE", "CANCEL", "", new EditText(context), context);
                    eventNAD.create();
                    eventNAD.ad.setOnShowListener(new DialogInterface.OnShowListener() {
                        @Override
                        public void onShow(DialogInterface dialogInterface) {

                            eventNAD.setBtnColor();
                            eventNAD.ad.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
                                @Override
                                public void onClick(View view) {

                                    if(eventNAD.et.getText().toString().equals("") || eventNAD.et.getText().toString().equals(null)) {

                                        Toast t = Toast.makeText(getApplicationContext(), "Please enter an event name", Toast.LENGTH_LONG);

                                        t.show();
                                    }

                                    else {

                                        Intent i = new Intent(MainActivity.this, ChooseGroupActivity.class);

                                        i.putExtra("eventName", eventNAD.et.getText().toString());

                                        famMain.close(true);

                                        eventNAD.ad.dismiss();

                                        startActivity(i);
                                    }
                                }
                            });
                        }
                    });

                    eventNAD.ad.show();
                }
            }
        });
    }

    private Cursor getAllGroups() {

        return mGroupDb.query(

                GroupContract.GroupEntry.TABLE_NAME,
                null,
                null,
                null,
                null,
                null,
                GroupContract.GroupEntry.COLUMN_GROUP_NAME
        );
    }

    private Cursor getAllEvents() {

        return mEventDb.query(

                EventContract.EventEntry.TABLE_NAME,
                null,
                null,
                null,
                null,
                null,
                EventContract.EventEntry.COLUMN_EVENT_NAME
        );
    }

    @Override
    public void onClick(int group_id) {

        Intent intent = new Intent(this, GroupDetailsActivity.class);

        intent.putExtra("groupID", group_id);

        Log.d(TAG, "DEBUG MA 202 " + group_id);

        startActivity(intent);
    }

    @Override
    public void onClickEvent(int event_id, int group_id) {

        Log.d(TAG, "DEBUG MA 369 ONCLICKEVENT " + event_id);

        Intent i = new Intent(MainActivity.this, EventDetailsActivity.class);

        Cursor cursor = mEventDb.query(
                EventContract.EventEntry.TABLE_NAME,
                null,
                EventContract.EventEntry._ID + " = " + event_id,
                null,
                null,
                null,
                EventContract.EventEntry.COLUMN_EVENT_DATE,
                null
        );

        i.putExtra("eventID", event_id);
        i.putExtra("groupID", group_id);

        Log.d(TAG, "DEBUG MA 372 " + group_id);

        startActivity(i);
    }

    public void updateNullView(RecyclerView.Adapter adapter, TextView textView) {

        Log.d(TAG, "DEBUG MA 427 " + adapter.getItemCount());

        if (adapter.getItemCount() == 0) {

            textView.setVisibility(View.VISIBLE);
        }

        else {

            textView.setVisibility(View.GONE);
        }
    }

    public long addNewGroup(String groupName) {

        Log.d(TAG, "DEBUG MDA 197 GROUP IS ADDED");

        ContentValues cv = new ContentValues();

        cv.put(GroupContract.GroupEntry.COLUMN_GROUP_NAME, groupName);

        return mGroupDb.insertOrThrow(GroupContract.GroupEntry.TABLE_NAME, null, cv);
    }
}

File: mobile/src/main/kotlin/com/planit/mobile/MainActivity.kt
package com.planit.mobile

import android.content.ContentValues
import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.os.Bundle
import android.os.Handler
import android.support.v4.app.FragmentActivity
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import android.widget.Toast.makeText

import com.github.clans.fab.FloatingActionButton
import com.github.clans.fab.FloatingActionMenu
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdView
import com.google.android.gms.ads.MobileAds
import com.planit.mobile.Adapters.GroupAdapter
import com.planit.mobile.Adapters.MemberAdapter
import com.planit.mobile.data.Contracts.EventContract
import com.planit.mobile.data.Contracts.GroupContract
import com.planit.mobile.data.Contracts.MemberContract
import com.planit.mobile.data.DbHelpers.MemberDbHelper
import com.planit.mobile.data.Useful
import com.planit.mobile.Adapters.EventAdapter
import com.planit.mobile.data.DbHelpers.EventDbHelper
import com.planit.mobile.data.DbHelpers.GroupDbHelper
import com.planit.mobile.R

import org.apache.commons.logging.LogFactory
import org.mortbay.jetty.Main
import org.w3c.dom.Text


//7.04 ONWARDS NEEDS TO BE DONE FOR EVENTS

class MainActivity : FragmentActivity(), GroupAdapter.GroupAdapterOnClickHandler, EventAdapter.EventAdapterOnClickHandler, MemberAdapter.MemberAdapterOnClickHandler {
    internal var context: Context = this

    private var eventDB: SQLiteDatabase? = null
    private var memberDB: SQLiteDatabase? = null

    private var mEventAdapter: EventAdapter? = null
    private var mMemberAdapter: MemberAdapter? = null

    private var mEventsList: RecyclerView? = null
    private var mMembersList: RecyclerView? = null

    private var eventCursor: Cursor? = null
    private var memberCursor: Cursor? = null
    internal lateinit var fabGroup: FloatingActionButton
    internal lateinit var fabDelete: FloatingActionButton

    private var mGroupDb: SQLiteDatabase? = null

    private var mGroupAdapter: GroupAdapter? = null

    private var mGroupsList: RecyclerView? = null

    private var groupCursor: Cursor? = null

    private var txtNullGroup: TextView? = null

    internal var TAG = "MainActivity"

    private var mAdView: AdView? = null

    private var expandedGroupId = -1

    internal var doubleBackToExitPressedOnce = false

    private val allGroups: Cursor
        get() = mGroupDb!!.query(

                GroupContract.GroupEntry.TABLE_NAME, null, null, null, null, null,
                GroupContract.GroupEntry.COLUMN_GROUP_NAME
        )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        mAdView = findViewById(R.id.m_ad)
        val adRequest = AdRequest.Builder().build()
        mAdView!!.loadAd(adRequest)

        MobileAds.initialize(this, "ca-app-pub-1144529456090164~2412831884")

        mGroupsList = findViewById<View>(R.id.m_rv_groups) as RecyclerView
        mGroupsList!!.isFocusable = false

        val groupLayoutManager = LinearLayoutManager(this)

        mGroupsList!!.layoutManager = groupLayoutManager

        val groupDbHelper = GroupDbHelper(this)

        mGroupDb = groupDbHelper.readableDatabase

        groupCursor = allGroups

        groupCursor!!.moveToFirst()

        txtNullGroup = findViewById<View>(R.id.m_tv_groups_null) as TextView

        mGroupsList!!.setHasFixedSize(true)

        mGroupAdapter = GroupAdapter(this, groupCursor, this)

        mGroupsList!!.adapter = mGroupAdapter

        updateNullView(mGroupAdapter!!, txtNullGroup)

        (findViewById<View>(R.id.m_fam_label) as FloatingActionMenu).open(true)

        fabGroup = findViewById(R.id.m_fab_add_group)
        fabDelete = findViewById(R.id.m_fab_delete)
        fabDelete.hide(false)

        fabDelete.setOnClickListener {
            if (expandedGroupId >= 0) {

                removeGroup()

                fabDelete.hide(true)
                famMain.close(true)

                groupCursor = allGroups
                groupCursor!!.moveToFirst()

                mGroupAdapter!!.refreshCursor(groupCursor)

                mGroupAdapter!!.notifyDataSetChanged()

                updateNullView(mGroupAdapter!!, txtNullGroup)

                expandedGroupId = -1
            }
        }

        famMain = findViewById<View>(R.id.m_fam) as FloatingActionMenu
        //famMain.setVisibility(View.GONE);

        fabGroup.setOnClickListener {
            Log.d(TAG, "DEBUG MA 137")

            val groupNAD = Useful.NameAlertDialog("Group", "CREATE", "CANCEL", "", EditText(context), context)
            groupNAD.create()
            groupNAD.ad.setOnShowListener {
                groupNAD.setBtnColor()
                groupNAD.ad.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener {
                    if (groupNAD.et.text.toString() == "" || groupNAD.et.text.toString() == null) {

                        val t = Toast.makeText(applicationContext, "Please enter a group name", Toast.LENGTH_LONG)

                        t.show()
                    } else {

                        addNewGroup(groupNAD.et.text.toString())

                        groupCursor = allGroups
                        groupCursor!!.moveToFirst()

                        mGroupAdapter!!.refreshCursor(groupCursor)

                        mGroupAdapter!!.notifyDataSetChanged()

                        updateNullView(mGroupAdapter!!, txtNullGroup)

                        expandedGroupId = -1

                        famMain.close(true)

                        groupNAD.ad.dismiss()
                    }
                }
            }

            groupNAD.ad.show()
        }

        findViewById<View>(R.id.m_fab_add_event).setOnClickListener {
            if (mGroupAdapter!!.itemCount == 0) {

                makeText(context, "Please create a group before creating an event", Toast.LENGTH_SHORT).show()
            } else if (expandedGroupId >= 0) {

                val eventNAD = Useful.NameAlertDialog("Event", "CREATE", "CANCEL", "", EditText(context), context)
                eventNAD.create()
                eventNAD.ad.setOnShowListener {
                    eventNAD.setBtnColor()
                    eventNAD.ad.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener {
                        if (eventNAD.et.text.toString() == "" || eventNAD.et.text.toString() == null) {

                            val t = Toast.makeText(applicationContext, "Please enter an event name", Toast.LENGTH_LONG)

                            t.show()
                        } else {

                            val i = Intent(this@MainActivity, EventDetailsActivity::class.java)
                            i.putExtra("eventName", eventNAD.et.text.toString())
                            i.putExtra("groupID", expandedGroupId)
                            i.putExtra("isNewEvent", true)

                            startActivity(i)

                            famMain.close(true)

                            eventNAD.ad.dismiss()

                            startActivity(i)
                        }
                    }
                }

                eventNAD.ad.show()
            }
        }

        findViewById<View>(R.id.m_fab_add_member).setOnClickListener {
            if (expandedGroupId >= 0) {

                val memberNAD = Useful.NameAlertDialog("Member", "ADD", "CANCEL", "", EditText(context), context)
                memberNAD.create()
                memberNAD.ad.setOnShowListener {
                    memberNAD.setBtnColor()

                    val btnPos = memberNAD.ad.getButton(DialogInterface.BUTTON_POSITIVE)

                    btnPos.setOnClickListener {
                        if (memberNAD.et.text.toString() == "" || memberNAD.et.text.toString() == null) {

                            val t = Toast.makeText(applicationContext, "Please enter a name", Toast.LENGTH_LONG)

                            t.show()
                        } else {

                            val i = Intent(this@MainActivity, MemberDetailsActivity::class.java)

                            i.putExtra("name", memberNAD.et.text.toString())
                            i.putExtra("groupID", expandedGroupId)

                            famMain.close(true)

                            memberNAD.ad.dismiss()

                            //saveGroupname();

                            startActivity(i)
                        }
                    }
                }

                memberNAD.ad.show()
            }
        }
    }

    override fun onBackPressed() {
        if (doubleBackToExitPressedOnce) {

            val intent = Intent(Intent.ACTION_MAIN)
            intent.addCategory(Intent.CATEGORY_HOME)
            startActivity(intent)
            return
        }

        this.doubleBackToExitPressedOnce = true
        Toast.makeText(this, "Please click BACK again to exit", Toast.LENGTH_SHORT).show()

        Handler().postDelayed({ doubleBackToExitPressedOnce = false }, 2000)
    }

    override fun onClick(group_id: Int, memberList: RecyclerView, eventList: RecyclerView, memberNull: TextView,
                         eventNull: TextView, llContent: LinearLayout, listGroupNameView: TextView, arrow: ImageView) {

        Log.d(TAG, "DEBUG MA 359")
        if (llContent.visibility != View.VISIBLE) {

            famMain.visibility = View.VISIBLE
            mMembersList = memberList
            mEventsList = eventList

            val memberLayoutManager = LinearLayoutManager(context)
            val eventLayoutManager = LinearLayoutManager(context)

            mMembersList!!.layoutManager = memberLayoutManager
            mEventsList!!.layoutManager = eventLayoutManager

            val memberDbHelper = MemberDbHelper(context)
            val eventDbHelper = EventDbHelper(context)

            memberDB = memberDbHelper.readableDatabase
            eventDB = eventDbHelper.readableDatabase

            memberCursor = getMembers(group_id)
            eventCursor = getEvents(group_id)

            memberCursor!!.moveToFirst()
            eventCursor!!.moveToFirst()

            mMembersList!!.setHasFixedSize(true)
            mEventsList!!.setHasFixedSize(true)

            mMemberAdapter = MemberAdapter(this, memberCursor, this, group_id)
            mEventAdapter = EventAdapter(this, eventCursor, this)

            mMembersList!!.adapter = mMemberAdapter
            mEventsList!!.adapter = mEventAdapter

            updateNullView(mMemberAdapter!!, memberNull)
            updateNullView(mEventAdapter!!, eventNull)

            //llContent.setVisibility(View.VISIBLE);

            if (expandedGroupId >= 0) {

                closingCollateral = true
                mGroupAdapter!!.hideExpandedGroup(expandedGroupId)
            }

            Log.d(TAG, "DEBUG MA 404")
            Useful.expand(llContent)

            //listGroupNameView.bringToFront();

            expandedGroupId = group_id

            Useful.rotate_down(this@MainActivity, arrow)

            Log.d(TAG, "DEBUG 363")
            famMain.open(true)
            fabDelete.show(true)
        } else {

            //llContent.setVisibility(View.GONE);

            //listGroupNameView.bringToFront();\

            Log.d(TAG, "DEBUG MA 377")

            if (!closingCollateral) {

                Useful.collapse(llContent, true)
                famMain.close(true)
                fabDelete.hide(true)
                //famMain.setVisibility(View.GONE);

                expandedGroupId = -1
            } else {

                Useful.collapse(llContent, false)
                closingCollateral = false
            }

            Useful.rotate_up(this@MainActivity, arrow)
        }
    }

    fun updateNullView(adapter: RecyclerView.Adapter<*>, textView: TextView?) {

        if (adapter.itemCount == 0) {

            textView!!.visibility = View.VISIBLE
        } else {

            textView!!.visibility = View.GONE
        }
    }

    fun addNewGroup(groupName: String): Long {

        val cv = ContentValues()

        cv.put(GroupContract.GroupEntry.COLUMN_GROUP_NAME, groupName)

        return mGroupDb!!.insertOrThrow(GroupContract.GroupEntry.TABLE_NAME, null, cv)
    }

    fun getMembers(groupID: Int): Cursor {

        //String[] selection = {MemberContract.MemberEntry.COLUMN_NAME};

        return memberDB!!.query(
                MemberContract.MemberEntry.TABLE_NAME, null,
                MemberContract.MemberEntry.COLUMN_GROUP_ID + "='" + groupID + "'", null, null, null, null
        )
    }

    fun getEvents(groupID: Int): Cursor {

        //String[] selection = {EventContract.EventEntry.COLUMN_EVENT_NAME};

        return eventDB!!.query(
                EventContract.EventEntry.TABLE_NAME, null,
                EventContract.EventEntry.COLUMN_GROUP_ID + "='" + groupID + "'", null, null, null, null
        )
    }

    override fun onClickEvent(event_id: Int, group_id: Int) {

        val i = Intent(context, EventDetailsActivity::class.java)

        //EVENT DB NOT NEEDED UNLIKE MAIN ACTIVITY BECAUSE GROUP ID IS ALREADY KNOWN (EVENT ID GIVEN AS PARAMETER)

        i.putExtra("eventID", event_id)
        i.putExtra("groupID", group_id)

        //saveGroupname();

        startActivity(i)
    }

    override fun onClick(name: String, memberID: Int, groupId: Int) {

        val intent = Intent(context, MemberDetailsActivity::class.java)

        intent.putExtra("name", name)

        intent.putExtra("groupID", groupId)
        intent.putExtra("firstMember", false)
        intent.putExtra("memberID", memberID)

        //saveGroupname();

        startActivity(intent)
    }

    fun saveGroupname(groupName: String, groupId: Int) {

        val cv1 = ContentValues()
        cv1.put(GroupContract.GroupEntry.COLUMN_GROUP_NAME, groupName)

        mGroupDb!!.update(GroupContract.GroupEntry.TABLE_NAME,
                cv1,
                GroupContract.GroupEntry._ID + "=" + groupId + "", null)
    }

    override fun onClickEdit(tvGroupName: TextView, groupId: Int) {

        val groupNAD = Useful.NameAlertDialog("Group", "DONE", "CANCEL", tvGroupName.text.toString(), EditText(this@MainActivity), this@MainActivity)
        groupNAD.create()
        groupNAD.ad.setOnShowListener {
            val btnPos = groupNAD.ad.getButton(DialogInterface.BUTTON_POSITIVE)
            groupNAD.setBtnColor()

            btnPos.setOnClickListener {
                val groupName = groupNAD.et.text.toString()
                tvGroupName.text = groupName
                saveGroupname(groupName, groupId)

                groupNAD.ad.dismiss()
            }
        }

        groupNAD.ad.show()
    }


    fun removeGroup(): Boolean {

        memberDB!!.delete(MemberContract.MemberEntry.TABLE_NAME, MemberContract.MemberEntry.COLUMN_GROUP_ID + "=" + expandedGroupId, null)
        eventDB!!.delete(EventContract.EventEntry.TABLE_NAME, EventContract.EventEntry.COLUMN_GROUP_ID + "=" + expandedGroupId, null)
        return mGroupDb!!.delete(GroupContract.GroupEntry.TABLE_NAME, GroupContract.GroupEntry._ID + "=" + expandedGroupId, null) > 0
    }

    companion object {

        lateinit var famMain: FloatingActionMenu
        var closingCollateral = false
    }
}
